hippod/api_object_post.py

```diff
@@ -75,85 +82,185 @@ def create_container_data_merge_issue_new(app, sha_sum, object_item, submitter):
+# init token reflects whether first subcontainer created or not
+def create_subcontainer_data_merge_issue_new(app, sha_major, sha_minor, object_item, submitter):
+    date = datetime.datetime.now().isoformat('T')
+    obj_root_path = app['DB_OBJECT_PATH']
+    cntr_path = os.path.join(obj_root_path, sha_major[0:2], sha_major, 'container.db')
+    d_sub = dict()
+    d_sub['submitter'] = submitter
+    d_sub['achievements'] = []
+    hippod.mime_data_db.save_object_item_data_list(app, object_item['object-item'])
+    d_sub['object-item'] = dict()
+    d_sub['object-item']['data'] = object_item['object-item']['data']
+    # del d_sub['object-item']['data']['size-real']
+    return json.dumps(d_sub, sort_keys=True,indent=4, separators=(',', ': '))
+
+
+def save_new_object_container(app, sha_major, sha_minor, object_item, submitter):
     obj_root_path = app['DB_OBJECT_PATH']
-    obj_root_pre_path = os.path.join(obj_root_path, sha_sum[0:2])
+    obj_root_pre_path = os.path.join(obj_root_path, sha_major[0:2])
     if not os.path.isdir(obj_root_pre_path):
         os.makedirs(obj_root_pre_path)
-    obj_root_full_path = os.path.join(obj_root_pre_path, sha_sum)
-    if not os.path.isdir(obj_root_full_path):
-        os.makedirs(obj_root_full_path)
-    achie_path = os.path.join(obj_root_full_path, 'achievements')
+    obj_root_major_path = os.path.join(obj_root_pre_path, sha_major)
+    if not os.path.isdir(obj_root_major_path):
+        os.makedirs(obj_root_major_path)
+    obj_minor_path = os.path.join(obj_root_major_path, sha_minor)
+    if not os.path.isdir(obj_minor_path):
+        os.makedirs(obj_minor_path)
+    achie_path = os.path.join(obj_minor_path, 'achievements')
     if not os.path.isdir(achie_path):
         os.makedirs(achie_path)
-    attachie_path = os.path.join(obj_root_full_path, 'attachments')
+    attachie_path = os.path.join(obj_root_major_path, 'attachments')
     if not os.path.isdir(attachie_path):
         os.makedirs(attachie_path)
 
-    file_path = os.path.join(obj_root_full_path, 'container.db')
-    if os.path.isfile(file_path):
+    file_path = os.path.join(obj_root_major_path, 'container.db')
+    minor_file_path = os.path.join(obj_minor_path, 'subcontainer.db')
+
+    if os.path.isfile(file_path) or os.path.isfile(minor_file_path):
         msg = "internal error: {}".format(inspect.currentframe())
         raise ApiError(msg)
     else:
-        cd = create_container_data_merge_issue_new(app, sha_sum, object_item, submitter)
+        cd = create_container_data_merge_issue_new(app, sha_major, sha_minor, object_item)
         fd = open(file_path, 'w')
         fd.write(cd)
         fd.close()
+        cd_minor = create_subcontainer_data_merge_issue_new(app, sha_major, sha_minor, object_item, submitter)
+        fd_minor = open(minor_file_path, 'w')
+        fd_minor.write(cd_minor)
+        fd_minor.close()
+
+
+def save_new_object_minor_container(app, sha_major, sha_minor, object_item, submitter):
```

Comment:
function is an candidate to be implemented in the the shared python file for DB handling